db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreQuery.java

```diff
@@ -0,0 +1,171 @@
+package org.commonjava.indy.cassandra.data;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.mapping.Mapper;
+import com.datastax.driver.mapping.MappingManager;
+import org.commonjava.indy.core.conf.IndyStoreManagerConfig;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.subsys.cassandra.CassandraClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.commonjava.indy.cassandra.data.CassandraStoreUtil.TABLE_STORE;
+
+@ApplicationScoped
+public class CassandraStoreQuery
```

Comment:
CassandraStoreDataManager has to override the query() method from the AbstractStoreDataManager, or this class will never be used. FYI

Reply:
After going through the code, I got what you mean. :) Actually the `CassandraStoreQuery` is not an implementation of `ArtifactStoreQuery`, just made the similar naming by coincidence, I put some query relates into cassandra session there.  So we still ref the `DefaultArtifactStoreQuery`.

Reply:
Sorry, yeah I just noticed that. If we still use DefaultArtifactStoreQuery though, won't it use a lot of method that stream all artifact stores, which will cause a lot of multi-node table scans?

Reply:
Yeah, I see what you mean. We need to refactor ArtifactStoreQuery to add more targeted method to avoid to invoke the getAllArtifactStores().  And we need to abandon the stream() way in the *StoreQuery side which depends on the getAllArtifactStores().